ement-room.el

```diff
@@ -2847,35 +2853,35 @@ MESSAGE is the message STRING was contained in."
         :then (lambda (data)
                 (when (buffer-live-p saved-buffer)
                   (with-current-buffer saved-buffer
-                    (ement-room--insert-preview message
+                    (ement-room--insert-preview event
                                                 related-event
-                                                num
+                                                link
                                                 (progn (unless (alist-get 'og:url data)
                                                          (push (cons 'og:url link) data))
                                                        (unless (alist-get 'og:title data)
                                                          (push (cons 'og:title "[untitled page]") data))
                                                        data)))))
         :else (lambda (why) (ement-debug "Error in fetching preview: " why))))))
 
-(defun ement-room--insert-preview (message related num data)
-  "Insert link preview for DATA into MESSAGE from node RELATED if it isn't already present.
-NUM is the index for this link."
+(defun ement-room--insert-preview (message related key data)
+  "Insert link preview for HTTP response DATA into MESSAGE from node RELATED,
+if it isn't already present. KEY will be the key used for inserting."
   (cond ((alist-get 'errcode data) (ement-debug "Rate limit in url preview"))
         (data (let ((related-node (ement-room--ewoc-node-before ement-ewoc
```

Comment:
Please avoid one-symbol-per-line formatting style.  This function has `(declare (indent defun))` specified to avoid this.  :)